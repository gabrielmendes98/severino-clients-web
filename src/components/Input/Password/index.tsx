import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import VisibilityIcon from '@mui/icons-material/Visibility';
import { useState } from 'react';
import IconButton from 'components/IconButton';
import Input, { InputProps } from '../index';
import InputAdornment from '../InputAdornment';

const PasswordInput = ({ children, InputProps, ...props }: InputProps) => {
  const [visible, setVisible] = useState(false);

  const preventMouseEvent = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
  };

  const toggleVisibility = () => setVisible(init => !init);

  return (
    <Input
      {...props}
      type={visible ? 'text' : 'password'}
      InputProps={{
        ...InputProps,
        endAdornment: (
          <InputAdornment position="end">
            <IconButton
              aria-label="mostrar ou esconder senha"
              onClick={toggleVisibility}
              onMouseDown={preventMouseEvent}
              onMouseUp={preventMouseEvent}
              edge="end"
            >
              {visible ? <VisibilityIcon /> : <VisibilityOffIcon />}
            </IconButton>
          </InputAdornment>
        ),
      }}
    >
      {children}
    </Input>
  );
};

export default PasswordInput;
