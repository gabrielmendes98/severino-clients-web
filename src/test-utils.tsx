import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
