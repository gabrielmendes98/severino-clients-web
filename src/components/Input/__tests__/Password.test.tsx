import { Form, Formik } from 'formik';
import { render, screen, userEvent, waitFor } from 'test-utils';
import Password from '../Password';

it('should match snapshot when hidden', () => {
  render(
    <Formik initialValues={{ password: '123123' }} onSubmit={jest.fn()}>
      <Form>
        <Password name="password" label="password" />
      </Form>
    </Formik>,
  );
  expect(screen.getByLabelText(/password/i)).toMatchSnapshot();
});

it('should match snapshot when showing', async () => {
  render(
    <Formik initialValues={{ password: '123123' }} onSubmit={jest.fn()}>
      <Form>
        <Password name="password" label="password" />
      </Form>
    </Formik>,
  );

  userEvent.click(screen.getByLabelText(/mostrar ou esconder senha/i));

  await waitFor(() => {
    expect(screen.getByLabelText('password')).toHaveAttribute('type', 'text');
  });

  expect(screen.getByLabelText(/password/i)).toMatchSnapshot();
});
