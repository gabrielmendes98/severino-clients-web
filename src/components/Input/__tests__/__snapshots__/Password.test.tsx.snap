// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when hidden 1`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
  id=":r0:"
  name="password"
  type="password"
  value="123123"
/>
`;

exports[`should match snapshot when showing 1`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
  id=":r1:"
  name="password"
  type="text"
  value="123123"
/>
`;
